src/ng/directive/ngModel.js

```diff
@@ -1095,141 +1251,97 @@ var DEFAULT_REGEXP = /(\s+|^)default(\s+|$)/;
+        var optionsCtrl = ctrls[0];
+        var parentOptions = ctrls[1] ? ctrls[1].$options : $modelOptions.defaultOptions;
+        optionsCtrl.$options = new ModelOptions(scope.$eval(attrs.ngModelOptions), parentOptions);
+      }
+    }
+  };
+}];
+
+
+/**
+ * @ngdoc service
+ * @name $modelOptions
+ * @description
  *
- * @example
+ * This service provides support to the {@link ngModelOptions} directive.
+ *
+ * Here, you can change the default settings from which {@link ngModelOptions}
+ * directives inherit.
+ *
+ * See the {@link ngModelOptions} directive for a list of the available options.
+ *
+ */
+function $ModelOptionsProvider() {
 
-  The following example shows how to override immediate updates. Changes on the inputs within the
-  form will update the model only when the control loses focus (blur event). If `escape` key is
-  pressed while the input field is focused, the value is reset to the value in the current model.
+  /**
+   * @ngdoc property
+   * @name $modelOptions#defaultOptions
+   * @type {Object}
+   * @description
+   * The default options to fall back on when there are no more ngModelOption
+   * directives as ancestors
+   *
+   * The initial default options are:
+   *
+   * * `updateOn`: `default`
+   * * `debounce`: `0`
+   */
+  var defaultOptions = {
+    updateOn: 'default',
+    debounce: 0
+  };
 
-  <example name="ngModelOptions-directive-blur" module="optionsExample">
-    <file name="index.html">
-      <div ng-controller="ExampleController">
-        <form name="userForm">
-          Name:
-          <input type="text" name="userName"
-                 ng-model="user.name"
-                 ng-model-options="{ updateOn: 'blur' }"
-                 ng-keyup="cancel($event)" /><br />
-
-          Other data:
-          <input type="text" ng-model="user.data" /><br />
-        </form>
-        <pre>user.name = <span ng-bind="user.name"></span></pre>
-      </div>
-    </file>
-    <file name="app.js">
-      angular.module('optionsExample', [])
-        .controller('ExampleController', ['$scope', function($scope) {
-          $scope.user = { name: 'say', data: '' };
-
-          $scope.cancel = function(e) {
-            if (e.keyCode == 27) {
-              $scope.userForm.userName.$rollbackViewValue();
-            }
-          };
-        }]);
-    </file>
-    <file name="protractor.js" type="protractor">
-      var model = element(by.binding('user.name'));
-      var input = element(by.model('user.name'));
-      var other = element(by.model('user.data'));
-
-      it('should allow custom events', function() {
-        input.sendKeys(' hello');
-        input.click();
-        expect(model.getText()).toEqual('say');
-        other.click();
-        expect(model.getText()).toEqual('say hello');
-      });
+  return {
+    defaultOptions: defaultOptions,
+    $get: function() {
+      return {
+        defaultOptions: new ModelOptions(defaultOptions)
```

Comment:
I have fixed it so that you can replace the `defaultOptions` object on the provider.
But if you start mucking about with that object in the service then it will still break.